test/e2e/adminapi_cluster_getlogs.go

```diff
@@ -0,0 +1,118 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+var _ = Describe("[Admin API] Kubernetes get pod logs action", func() {
+	BeforeEach(skipIfNotInDevelopmentEnv)
+
+	const containerName = "e2e-test-container-name"
+	const podName = "e2e-test-pod-name"
+
+	When("in a standard openshift namespace", func() {
+		const namespace = "default"
+
+		It("should be able to get logs from a container of a pod", func() {
+			testGetPodLogsOK(containerName, podName, namespace)
+		})
+	})
+
+	When("in a customer namespace", func() {
+		const namespace = "e2e-test-namespace"
+
+		It("should be not be able to get logs from customer workload namespaces", func() {
+			testGetPodLogsFromCustomerNamespaceForbidden(containerName, podName, namespace)
+		})
+	})
+})
+
+// We will create a pod with known logs of its container and will compare the logs gotten through the kubernetes client and through the Admin API.
+func testGetPodLogsOK(containerName, podName, namespace string) {
+	By("creating a pod in default namespace with some known logs then comparing those logs with the logs of the container in the pod created")
+	pod := mockPod(containerName, podName, namespace)
+	pod, err := clients.Kubernetes.CoreV1().Pods(namespace).Create(context.Background(), pod, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		err = clients.Kubernetes.CoreV1().Pods(namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{})
+		Expect(err).NotTo(HaveOccurred())
+	}()
+	//  The below loop waits of the phase of the created pod to be Succeeded, if not it times out after 60 seconds and returns error.
+	for i := 0; pod.Status.Phase != corev1.PodSucceeded; i++ {
+		if i == 60 {
+			err = api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeInternalServerError, "", "")
+			Expect(err).NotTo(HaveOccurred())
+		}
+		pod, err = clients.Kubernetes.CoreV1().Pods(namespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		time.Sleep(time.Second)
+	}
```

Comment:
Take a look at `k8s.io/apimachinery/pkg/util/wait` package: there are `PollImmediate`, `PollImmediateUntil` and others.

Reply:
+1

Reply:
okay used `wait.Poll instead of my for loop.